matching_brackets.exs

```diff
@@ -0,0 +1,49 @@
+defmodule BracketsBalance do
+  @brackets %{
+    "{" => "}",
+    "(" => ")",
+    "[" => "]"
+  }
+
+  def add_to_stack(stack, counter, check_string) do
+    if counter < length(check_string) do
+      stack =
+        if Map.has_key?(@brackets, Enum.at(check_string, counter)) do
+          stack ++ [Enum.at(check_string, counter)]
+        else
+            if Enum.at(check_string, counter) == @brackets[Enum.at(stack, -1)] do
+                stack |> Enum.reverse() |> tl() |> Enum.reverse()
+            else
+                if Map.values(@brackets) |> Enum.member?(Enum.at(check_string, counter)) do
+                    [Enum.at(check_string, counter)]++stack
```

Comment:
You've found an unmatched closing bracket here - it's better to return `:error` as fast as possible in this case, instead of iterating through the rest of the string.